peerdroid/src/main/java/rdx/works/peerdroid/domain/ConnectorExtensionExchangeInteraction.kt

```diff
@@ -0,0 +1,19 @@
+package rdx.works.peerdroid.domain
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+sealed class ConnectorExtensionExchangeInteraction(
+    @SerialName("discriminator")
+    val discriminator: String
+) {
+
+    @Serializable
+    data class LinkClient(
+        @SerialName("publicKey")
+        val publicKey: String,
+        @SerialName("signature")
+        val signature: String
```

Comment:
I'd have to define custom serializers to make this change and I want to understand what would be the benefits, as I see this class as a DTO, with no other purpose than being serialized to JSON and sent to CE. Usually, I choose to have primitive types inside DTOs as I have a better sense of control when building it, for example by having a mapper called manually and throwing an exception if a field is not of the expected format, rather than having this kind of logic inside the serializer, somewhat decoupled of the code that is building the DTO, therefore harder to find/debug/understand, on the other hand, it means duplicating the mapping logic. I'm curious about your experience with custom serializers so far and if we agree that we're going to use them and enforce custom types from now on inside all the DTOs.

Reply:
Meanwhile, by looking at the gateway-related code I, found out that the preferred approach on the project is to use custom serializers and I think my point is not valid anymore, I didn't know and I thought it was up for discussion, but it makes sense to continue using serializers for consistency and for the benefit of not having custom type to primitives mapping logic duplicated. I added PublicKey and Signature serializers to the core module.